includes/admin/class-amp-post-meta-box.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * AMP meta box settings.
+ *
+ * @package AMP
+ * @since 0.6
+ */
+
+/**
+ * Post meta box class.
+ *
+ * @since 0.6
+ */
+class AMP_Post_Meta_Box {
+
+	/**
+	 * Assets handle.
+	 *
+	 * @const string
+	 */
+	const ASSETS_HANDLE = 'amp-post-meta-box';
+
+	/**
+	 * Initialize.
+	 *
+	 * @since 0.6
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_assets' ) );
+	}
+
+	/**
+	 * Enqueue admin assets.
+	 *
+	 * @since 0.6
+	 */
+	public function enqueue_admin_assets() {
```

Comment:
I am not sure if I am missing something but from my perspective it is not loading on all pages since it checks if the `$post->type` is set and the `post_supports_amp()`. It does load on the `edit.php`, `post.php` and `post-new.php` if the post type as AMP support. That said it isn't really necessary to load it on `edit.php` so I will add a check to strictly load it on `post.php` and `post-new.php`.

Reply:
@ThierryA oh yes, I see you're right. and that it wouldn't get enqueued. But it would get enqueued if an admin screen happened to define a global `$post` and didn't clean it up. So yeah, I think it is best to explicitly check if the `current_screen` is the post editor. Maybe `'post' === get_current_screen()->base` would be a good check?